Doc: Add PodsReadyRequeueStrategy documentation (#1727)
kubernetes-sigs/kueue: site/content/en/docs/tasks/setup_sequential_admission.md

    waitForPodsReady:
      enable: true

>       timeout: 10m

Reviewer: can you add `blockAdmission` as well?

Author: Sure.

Author: Done.

The `blockAdmission` (`waitForPodsReady.blockAdmission`) is an optional parameter, defaulting to `false`.
When the `blockAdmission` is set to `true` regardless of the `enable` (`waitForPodsReady.enable`) is set to `false`,
the `enable` is overridden to `true`.
When the `blockAdmission` is set to `true`, admitted Workload with not ready pods block admission for other Workloads.

### Requeuing Strategy
{{% alert title="Warning" color="warning" %}}
_Available in Kueue v0.6.0 and later_
{{% /alert %}}

The `requeuingStrategy` (`waitForPodsReady.requeuingStrategy`) contains optional parameters: 
`timestamp` (`waitForPodsReady.requeuingStrategt.timestamp`) and `backoffLimitCount` (`waitForPodsReady.requeuingStrategt.backoffLimitCount`).

The `timestamp` field defines which timestamp Kueue uses to order the Workloads in the queue:

- `Eviction` (default): The `lastTransitionTime` of the `Evicted=true` condition with `PodsReadyTimeout` reason in a Workload.
- `Creation`: The creationTimestamp in a Workload.

If you want to re-queue a Workload evicted by the `PodsReadyTimeout` back to its original place in the queue,

> you should set the `Creation` to the timestamp.

Reviewer: ```suggestion
you should set the timestamp to `Creation`.
```

Author: Done.

`PodsReady=False`), then the Workload's admission is
cancelled, the corresponding job is suspended and the Workload is requeued.

> ### ReQueuing Strategy

Reviewer: ```suggestion
### Requeuing Strategy
```

Author: Done.

{{% alert title="Warning" color="warning" %}}
_Available in Kueue v0.6.0 and later_
{{% /alert %}}

> The requeuingStrategy (`waitForPodsReady.requeuingStrategy`) is optional parameters, 

Reviewer: ```suggestion
The requeuingStrategy (`waitForPodsReady.requeuingStrategy`) contains optional parameters:
`timestamp` and `backoffLimitCount`.
```

Author: Done.

timestamp (`waitForPodsReady.requeuingStrategt.timestamp`) is defaulting to `Eviction`, and

> backoffLimitCount (`waitForPodsReady.requeuingStrategt.backoffLimitCount`) is defaulting to null.

Reviewer: remove these lines. It could be explained below.

Author: Done.

> When you specify the timestamp, a Workload uses the below time to order the Workloads in the queue:

Reviewer: ```suggestion
The `timestamp` field defines which timestamp Kueue uses to order the Workloads in the queue:
```

Author: Done.

> - `Eviction`: The `lastTransitionTime` of the `Evicted=true` condition with `PodsReadyTimeout` reason in a Workload.

Reviewer: ```suggestion
- `Eviction` (default): The `lastTransitionTime` of the `Evicted=true` condition with `PodsReadyTimeout` reason in a Workload.
```

Author: Done.

- `Creation`: The creationTimestamp in a Workload.

> If you want to re-queue a Workload evicted by the `PodsReadyTimeout` to head of the queue,

Reviewer: ```suggestion
If you want to re-queue a Workload evicted by the `PodsReadyTimeout` back to its original place in the queue,
```

Author: Done.

> you can set the `Creation` to the timestamp. 

Reviewer: ```suggestion
you should use the `Creation` to the timestamp.
```

Author: Done.

> A Workload evicted by the `PodsReadyTimeout` reason is re-queued until the count is reached to backoffLimitCount.

Reviewer: ```suggestion
Kueue will requeue a Workload evicted by the `PodsReadyTimeout` reason until the number of requeues reaches `backoffLimitCount`.
```

Author: Done.

> If you don't specify any integer for backoffLimitCount, 

Reviewer: ```suggestion
If you don't specify any value for backoffLimitCount, 
```

Author: Done.

a Workload is repeatedly and endlessly re-queued to the queue based on timestamp.

> After the count is reached, a Workload is [deactivated](docs/concepts/workload/#active).

Reviewer: ```suggestion
Once the the number of requeues reaches the limit, Kueue [deactivates the Workload](docs/concepts/workload/#active).
```

Author: Done.
